packages/effection/src/duplex-channel.ts

```diff
@@ -0,0 +1,29 @@
+import { subscribe, ChainableSubscribable } from '@effection/subscription';
+import { Channel } from '@effection/channel';
+
+export interface Sendable<T> {
+  send(message: T): void;
+}
+
+export type DuplexChannel<S, R, RReturn = undefined> = ChainableSubscribable<R, RReturn> & Sendable<S>;
+
+export type DuplexPair<Tx, Rx> = [DuplexChannel<Tx, Rx>, DuplexChannel<Rx, Tx>];
```

Comment:
It's common in JS to make things like this destructurable via both name and index. E.g.

```ts
interface DuplexPair<Tx,Rx> extends IterableIterator<[Tx,Rx]> {
  local: DuplexChannel<Tx,Rx>;
  remote: DuplexChannel<Rx,Tx>;
}
```

That way both of the following work.

```ts
let { local, remote } = DuplexChannel<Send,Receive>();
let [tx,rx] = DuplexChannel<Send,Receive>();
```
Not necessary here, but something to consider when we move it over to effection proper